ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -231,6 +261,42 @@ private HelixBootstrapUpgradeUtil(String hardwareLayoutPath, String partitionLay
     }
   }
 
+  /**
+   * Generate the partition override map containing partition state from all datacenters.
+   */
+  private Map<String, Map<String, String>> generatePartitionOverrideFromAllDCs() {
+    Map<String, Map<String, String>> partitionOverrideInfos = new HashMap<>();
+    for (PartitionId partitionId : staticClusterMap.getAllPartitionIds(null)) {
+      String partitionName = partitionId.toPathString();
+      Map<String, String> partitionProperties = new HashMap<>();
+      partitionProperties.put(ClusterMapUtils.PARTITION_STATE,
+          partitionId.getPartitionState() == PartitionState.READ_WRITE ? ClusterMapUtils.READ_WRITE_STR
+              : ClusterMapUtils.READ_ONLY_STR);
+      partitionOverrideInfos.put(partitionName, partitionProperties);
+    }
+    return partitionOverrideInfos;
+  }
+
+  /**
+   * Uploads the seal state of all partitions in the format of map.
+   */
+  private void uploadPartitionOverride(Map<String, Map<String, String>> partitionOverrideInfos) {
+    Properties storeProps = new Properties();
+    storeProps.setProperty("helix.property.store.root.path", "/" + clusterName);
+    HelixPropertyStoreConfig propertyStoreConfig = new HelixPropertyStoreConfig(new VerifiableProperties(storeProps));
+    info("Setting partition override for all datacenters.");
+    for (Map.Entry<String, ClusterMapUtils.DcZkInfo> entry : dataCenterToZkAddress.entrySet()) {
```

Comment:
So the same information is going to be duplicated in every data center? Do you think that is the right thing to do?

Reply:
The ClusterManager on server only talks to zookeeper in local datacenter to get the override info. Hence, it's necessary to send the override info to all DCs. Also, in the future, the override info may vary by DCs, we should upload the specific override info to partitular DC separately.